insights/parsers/tests/test_ip.py

```diff
@@ -352,6 +352,110 @@ def test_ip_data_Link():
+    RX: bytes  packets  mcast   bcast
+    2          0        0       0
+    TX: bytes  packets
+    2          0
+    vf 3 MAC 5a:05:77:0e:00:7f, spoof checking on, link-state auto, trust off
+    RX: bytes  packets  mcast   bcast
+    3          0        0       0
+    TX: bytes  packets
+    3          0
+    vf 4 MAC 6a:5c:04:27:60:16, spoof checking on, link-state auto, trust off
+    RX: bytes  packets  mcast   bcast
+    4          0        0       0
+    TX: bytes  packets
+    4          0
+    vf 5 MAC 22:3f:6a:a4:dc:b1, spoof checking on, link-state auto, trust off
+    RX: bytes  packets  mcast   bcast
+    5          0        0       0
+    TX: bytes  packets
+    5          0
+    vf 6 MAC 86:a3:17:a3:46:b9, spoof checking on, link-state auto, trust off
+    RX: bytes  packets  mcast   bcast
+    6          0        0       0
+    TX: bytes  packets
+    6          0
+4: p5p2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
+    link/ether b4:96:91:17:10:2f brd ff:ff:ff:ff:ff:ff promiscuity 0 addrgenmode eui64 numtxqueues 8 numrxqueues 8 gso_max_size 65536 gso_max_segs 65535
+    RX: bytes  packets  errors  dropped overrun mcast
+    0          0        0       0       0       0
+    TX: bytes  packets  errors  dropped carrier collsns
+    0          0        0       0       0       0
+    vf 0 MAC 2a:2a:69:f5:84:b4, spoof checking on, link-state auto, trust off
+    RX: bytes  packets  mcast   bcast
+    0          0        0       0
+    TX: bytes  packets
+    0          0
+    vf 1 MAC 86:2c:59:c1:ac:da, spoof checking on, link-state auto, trust off
+    RX: bytes  packets  mcast   bcast
+    1          0        0       0
+    TX: bytes  packets
+    1          0
+    vf 2 MAC ee:2a:1a:15:6e:5b, spoof checking on, link-state auto, trust off
+    RX: bytes  packets  mcast   bcast
+    2          0        0       0
+    TX: bytes  packets
+    2          0
+    vf 3 MAC 3a:41:95:30:6b:63, spoof checking on, link-state auto, trust off
+    RX: bytes  packets  mcast   bcast
+    3          0        0       0
+    TX: bytes  packets
+    3          0
+    vf 4 MAC c2:40:c0:ee:9d:28, spoof checking on, link-state auto, trust off
+    RX: bytes  packets  mcast   bcast
+    4          0        0       0
+    TX: bytes  packets
+    4          0
+    vf 5 MAC de:61:ec:a6:84:de, spoof checking on, link-state auto, trust off
+    RX: bytes  packets  mcast   bcast
+    5          0        0       0
+    TX: bytes  packets
+    5          0
+    vf 6 MAC 3a:83:da:6f:00:0c, spoof checking on, link-state auto, trust off
+    RX: bytes  packets  mcast   bcast
+    6          0        0       0
+    TX: bytes  packets
+    6          0
+""".strip()
+
+
+def test_ip_link_for_virtual_functions():
+    output = ip.IpLinkInfo(context_wrap(IP_S_LINK_VF)).data
+    assert 'p5p2' in output
+    assert output.get('p5p2').data.get('vf')[2].get('tx_bytes') == 2
```

Comment:
@psachin - Please add more unit tests for non-zero stats
e.g
~~~
    assert output.get('p5p2').data.get('vf')[2].get('tx_bytes') == 2
    assert output.get('p5p2').data.get('vf')[0].get('rx_bytes') == 10000
    assert output.get('p5p2').data.get('vf')[0].get('tx_packets') == 12030
    assert output.get('p5p2').data.get('vf')[2].get('tx_packets') == 900
    assert output.get('p5p2').data.get('vf')[2].get('rx_mcast') == 10
~~~
Please adjust your sample data accordingly.

Reply:
Sure. Done.